helper/helper.go

```diff
@@ -291,7 +257,7 @@ func DoCleanup() {
 			logVerbose("Encountered error closing error file: %v", err)
 		}
 	}
-	err := flushAndCloseRestoreWriter("Current writer pipe on cleanup", 0)
+	err := Restorer.flushAndCloseRestoreWriter("Current writer pipe on cleanup", 0)
```

Comment:
```suggestion
	err := FlushAndCloseRestoreWriter("Current writer pipe on cleanup", 0)
```

Reply:
This is a rollback for change allowing extra check for opened pipes.

Reply:
Why do we need a global object from the restore when backuping?

Reply:
At this place refactoring has not went so far to fix initial design flaw. DoCleanup() is a common function used for both backup and restore and it called FlushAndCloseRestoreWriter() from restore_helper. Here it just replaced with the object which allows to be mocked.
I agree that it looks strange and I'd like to change it too, but let's limit changes for this work.

Reply:
That's exactly what I suggest not to change the current behavior, when the helper calls the flush and close function without any objects. The patch simply moves it to another file and therefore it should be capitalized.

Reply:
And it can still be mocked! See the [comment](https://github.com/arenadata/gpbackup/pull/102#discussion_r1792842271).

Reply:
Okay, let's postpone this to the next refactoring.